components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -482,8 +482,16 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
 
         try {
 
-            boolean exist = tenantGovReg.resourceExists(applicationNode);
-            if (!exist) {
+            boolean appNodeExists = tenantGovReg.resourceExists(applicationNode);
+
+            if (ArrayUtils.isEmpty(permissions)) { // no new permissions
```

Comment:
Lets follow the following format when commenting
- Start with a capital letter
- Ends with a full stop.

ex: `// No new permissions.`

Look for other places as well

Reply:
And let's make the comment a bit more descriptive and add it above this line

Reply:
Outdated. `updatePermissions` method has been refactored in a later change and the comments are no longer applicable.